packages/scaffold-ui/src/modal/w3m-modal/index.ts

```diff
@@ -69,6 +69,9 @@ export class W3mModalBase extends LitElement {
 
   @state() private padding = vars.spacing[1]
 
+  @state() private mobileFullScreen =
+    OptionsController.state.enableMobileFullScreen && CoreHelperUtil.isMobile()
```

Comment:
Can we move this `CoreHelperUtil.isMobile()` check to Controller instead? So we could prevent setting this flag to true if is not mobile, and no need to maintain on the UI